src/software/thunderscope/robot_diagnostics/controller_diagnostics.py

```diff
@@ -0,0 +1,245 @@
+            )
+
+    def __parse_move_event_value(self, value, factor):
+        if abs(value) < (DEADZONE_PERCENTAGE * factor):
+            return 0
+        else:
+            return value / (XBOX_MAX_RANGE * factor)
+
+    def __parse_dribbler_event_value(self, value):
+        # can probably combine these 2 parsers but abastracting the +/- is annoying so 2 funcs it is
+        new_speed = self.dribbler_speed - value * DRIBBLER_STEPPER
+        if MIN_DRIBBLER_RPM < new_speed < MAX_DRIBBLER_RPM:
+            return new_speed
+
+    def __parse_kick_event_value(self, value) -> int:
+        new_power = self.power + value * POWER_STEPPER
+        if MIN_POWER < new_power < MAX_POWER:
+            return new_power
+
+    def __parse_dribbler_enabled_event_value(self, value) -> bool:
+        return value > (XBOX_BUTTON_MAX_RANGE / 2)
+
+    def __send_move_command(self):
+        motor_control = MotorControl()
+
+        motor_control.direct_velocity_control.velocity.x_component_meters = self.x_vel
+        motor_control.direct_velocity_control.velocity.y_component_meters = self.y_vel
+        motor_control.direct_velocity_control.angular_velocity.radians_per_second = (
+            self.ang_vel
+        )
+
+        motor_control.dribbler_speed_rpm = 0
+
+        # if self.enable_dribbler:
+        #     motor_control.dribbler_speed_rpm = dribbler_speed
+        # maybe just use indefinite instead? or have setting to turn on 'smooth scrolling'
+        if self.dribbler_enabled:
+            motor_control.dribbler_speed_rpm = (
+                self.constants.indefinite_dribbler_speed_rpm
+            )
+
+        logging.info("Sending motor control: " + motor_control)
+
+        self.proto_unix_io.send_proto(MotorControl, motor_control)
+
+    def __send_kick_command(self):
+        power_control = PowerControl()
+        power_control.geneva_slot = 1
+        power_control.chicker.kick_speed_m_per_s = self.kick_power
+
+        logging.info("Sending kick power control: " + power_control)
+
+        self.proto_unix_io.send_proto(PowerControl, power_control)
+
+    def __send_chip_command(self):
+        power_control = PowerControl()
+        power_control.geneva_slot = 1
+        power_control.chicker.chip_distance_meters = (
+            self.kick_power
+        )  # not sure if we should use this value
+
+        logging.info("Sending chip power control: " + power_control)
+
+        self.proto_unix_io.send_proto(PowerControl, power_control)
+
+    def __process_event(self, event):
+        logging.info("Processing event: " + str(event))
+
+        abs_event = categorize(event)
+        event_t = ecodes.bytype[abs_event.event.type][abs_event.event.code]
+        # TODO: nump python version so we can use pattern matching for this
+        # TODO: make codebase use python logging instead of stdout
+        logging.info("Processing event: " + str(event_t))
```

Comment:
reminder to remove the logs once the PR is cleaned up and ready to be merged.

Reply:
forgot to add the issue about Thunderscope logging - my idea was that it would be reduced a level to `DEBUG` or `TRACE`, it may be useful for debugging new controllers that aren't Xbox